src/main/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacade.java

```diff
@@ -399,19 +415,25 @@ public final Response promoteUserFromWaitingList(@Context final HttpServletReque
                                                      @PathParam("event_id") final String eventId,
                                                      @PathParam("user_id") final Long userId, final Map<String, String> additionalInformation) {
         try {
-            if (!isUserAnAdminOrEventManager(userManager, request)) {
+            if (!isUserAbleToManageEvents(userManager, request)) {
                 return new SegueErrorResponse(Status.FORBIDDEN, "You must be a staff user to access this endpoint.")
                     .toResponse();
             }
 
+            RegisteredUserDTO currentUser = this.userManager.getCurrentRegisteredUser(request);
```

Comment:
Can't remember if there were other methods that would take the user to check if you already had it? I guess this currently will look up the same user info from the db twice (once for the role check and once to get the current user on line 423). Not hugely important in the grand scheme of things though.